cmd/translate/main_grpc_integration_test.go

```diff
@@ -672,6 +674,272 @@ func Test_ListTranslations_gRPC(t *testing.T) {
+	createTranslation(ctx, t, service2.GetId(), &translatev1.Translation{Original: false, Language: langs[2].String()})
+
+	service3 := createService(ctx, t)
+	createTranslation(ctx, t, service2.GetId(), &translatev1.Translation{Original: false, Language: langs[3].String()})
+
+	service4 := createService(ctx, t)
+	createTranslation(ctx, t, service2.GetId(), &translatev1.Translation{Original: false, Language: langs[4].String()})
+
+	service5 := createService(ctx, t)
+	service6 := createService(ctx, t)
+	service7 := createService(ctx, t)
+	service8 := createService(ctx, t)
+
+	tests := []struct {
+		request     *translatev1.UpdateTranslationRequest
+		translation *translatev1.Translation
+		name        string
+	}{
+		{
+			name: "Update messages field in original translation",
+			request: randUpdateTranslationReq(t, service.GetId(), &translatev1.Translation{
+				Language: langs[0].String(),
+				Messages: []*translatev1.Message{
+					{
+						Id: "Hello", Message: "World",
+					},
+				},
+			},
+				&field_mask.FieldMask{Paths: []string{"messages"}}),
+			translation: createTranslation(ctx, t, service.GetId(), &translatev1.Translation{
+				Original: true,
+				Language: langs[0].String(),
+				Messages: []*translatev1.Message{
+					{
+						Id: "Hello", Message: "Planet",
+					},
+				},
+			}),
+		},
+		{
+			name: "Add message to original translation",
+			request: randUpdateTranslationReq(t, service2.GetId(), &translatev1.Translation{
+				Language: langs[0].String(),
+				Messages: []*translatev1.Message{
+					{
+						Id: "Bye", Message: "World",
+					},
+				},
+			},
+				&field_mask.FieldMask{Paths: []string{"messages"}}),
+			translation: createTranslation(ctx, t, service2.GetId(), &translatev1.Translation{
+				Original: true,
+				Language: langs[0].String(),
+				Messages: []*translatev1.Message{
+					{
+						Id: "Hello", Message: "Planet",
+					},
+				},
+			}),
+		},
+		{
+			name: "Add message to non original translation",
+			request: randUpdateTranslationReq(t, service3.GetId(), &translatev1.Translation{
+				Language: langs[0].String(),
+				Messages: []*translatev1.Message{
+					{
+						Id: "Bye", Message: "World",
+					},
+				},
+			},
+				&field_mask.FieldMask{Paths: []string{"messages"}}),
+			translation: createTranslation(ctx, t, service3.GetId(), &translatev1.Translation{
+				Original: false,
+				Language: langs[0].String(),
+				Messages: []*translatev1.Message{
+					{
+						Id: "Hello", Message: "Planet",
+					},
+				},
+			}),
+		},
+		{
+			name: "Update messages field in non original translation",
+			request: randUpdateTranslationReq(t, service4.GetId(), &translatev1.Translation{
+				Language: langs[0].String(),
+				Messages: []*translatev1.Message{
+					{
+						Id: "Hello", Message: "World",
+					},
+				},
+			},
+				&field_mask.FieldMask{Paths: []string{"messages"}}),
+			translation: createTranslation(ctx, t, service4.GetId(), &translatev1.Translation{
```

Comment:
All tests MUST be isolated - no resource sharing. Any reason why you use the same service as in the previous test?

Reply:
I will fix it. 👍